frontend/views/containers/contributions/SupportHistory.vue

```diff
@@ -26,19 +29,58 @@ export default ({
       history: []
     }
   },
+  mixins: [PaymentsMixin],
   components: {
     BarGraph
   },
-  created () {
-    // Todo replace history with real data
-    const testNumber = 6
-    for (let i = testNumber; i > 0; i--) {
-      const date = new Date()
-      date.setMonth(date.getMonth() - i)
-      this.history.push({
-        total: 1 / testNumber * (testNumber - i + 1),
-        title: humanDate(date, { month: 'long' })
+  computed: {
+    ...mapGetters([
+      'groupSettings',
+      'currentPaymentPeriod',
+      'periodBeforePeriod'
+    ]),
+    mincome () {
+      return this.groupSettings.mincomeAmount
+    },
+    periods () {
+      const periods = [this.currentPaymentPeriod]
+      for (let i = 0; i < MAX_HISTORY_PERIODS - 1; i++) {
+        periods.unshift(this.periodBeforePeriod(periods[0]))
+      }
+      return periods
+    }
+  },
+  mounted () {
+    this.updateHistory()
+  },
+  methods: {
+    parsePayments (payments) {
+      const list = {}
+      payments.forEach(payment => {
+        const { from, to, amount } = payment
+        list[from] = (list[from] || 0) + amount
+        list[to] = (list[to] || 0) - amount
       })
+      return {
+        numReceivers: Object.values(list).filter(amount => amount < 0).length,
+        totalDistributionAmount: Object.values(list).filter(amount => amount > 0).reduce((total, curValue) => total + curValue, 0)
+      }
+    },
+    async updateHistory () {
+      this.history = await Promise.all(this.periods.map(async (period, i) => {
```

Comment:
OK, either in this function, or in `this.periods`, we need to show 0% for months that don't have payments for them (or don't even exist).

Remember to test locally before pushing, since a quick test on my side shows this when first starting up a group with 3 users and no payments:

![Screen Shot 2022-10-26 at 4 49 37 PM](https://user-images.githubusercontent.com/138706/198159658-5a92b6a9-d385-48ff-8a3c-15157f6c19ef.jpg)

Reply:
The following screenshot shows the example of support history graph for the last six months when I add payments only 2 months.
![image](https://user-images.githubusercontent.com/57976479/198165937-86c86bd2-0294-47b5-ad10-9454d82b35b5.png)

Reply:
Seeing multiple months of red "0%" will be discouraging for new groups. I think there should be some other placeholder for non-existent months, so I've made a comment on Figma: https://www.figma.com/file/mxGadAHfkWH6qApebQvcdN?node-id=1712:413#294246554

(Leaving this comment here because it's relevant, not because it should be blocking the PR)